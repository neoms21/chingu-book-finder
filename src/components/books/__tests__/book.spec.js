import { shallow } from 'enzyme';
import React from 'react';
import Book from '../book';
import 'jest-styled-components';

describe('Book tests', () => {
  let defaultProps = { book: { volumeInfo: { title: 'title', authors: ['aa'] } } };

  it('book component is loaded correctly without images', () => {
    const wrapper = shallow(<Book {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('book component is loaded correctly with images', () => {
    defaultProps.book.volumeInfo.imageLinks = { thumbnail: '' };
    const wrapper = shallow(<Book {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('book component is loaded if no authors', () => {
    defaultProps.book.volumeInfo.authors = undefined;
    const wrapper = shallow(<Book {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
