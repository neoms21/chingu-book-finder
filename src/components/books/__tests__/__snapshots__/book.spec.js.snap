// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book tests book component is loaded correctly with images 1`] = `
<styled.div>
  <styled.img
    alt="title"
    src=""
  />
  <styled.div>
    <styled.h4>
      title
    </styled.h4>
    <styled.div>
      <span
        key="aa"
      >
        aa
      </span>
    </styled.div>
    <styled.div>
      <styled.span>
         
        <i
          className="icon star"
        />
      </styled.span>
      <a>
        <Icon
          as="i"
          name="external alternate"
        />
      </a>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Book tests book component is loaded correctly without images 1`] = `
<styled.div>
  <styled.img
    alt="title"
    src="no-img.jpg"
  />
  <styled.div>
    <styled.h4>
      title
    </styled.h4>
    <styled.div>
      <span
        key="aa"
      >
        aa
      </span>
    </styled.div>
    <styled.div>
      <styled.span>
         
        <i
          className="icon star"
        />
      </styled.span>
      <a>
        <Icon
          as="i"
          name="external alternate"
        />
      </a>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Book tests book component is loaded if no authors 1`] = `
<styled.div>
  <styled.img
    alt="title"
    src=""
  />
  <styled.div>
    <styled.h4>
      title
    </styled.h4>
    <styled.div>
      <styled.span>
         
        <i
          className="icon star"
        />
      </styled.span>
      <a>
        <Icon
          as="i"
          name="external alternate"
        />
      </a>
    </styled.div>
  </styled.div>
</styled.div>
`;
